chia/server/node_discovery.py

```diff
@@ -319,9 +349,9 @@ async def _connect_to_peers(self, random) -> None:
                         addr = None
                         continue
                     # only consider very recently tried nodes after 30 failed attempts
-                    # attempt a node once per 2 hours if we lack connections to increase the chance
+                    # attempt a node once per 30 minutes if we lack connections to increase the chance
                     # to try all the peer table.
-                    if now - info.last_try < 2 * 3600 and tries < 30:
+                    if now - info.last_try < 1800 and tries < 30:
```

Comment:
what would you think about:
```
if now - info.last_try < 1800 + tries * 360 and tries < 30:
```

30 seems like a very large number of attempts per peer, but I suppose we really want peers to be "sticky" in our peer db, in case they go offline and come back again later. this has to be weighed against retrying a peer often.

If we just increase the retry frequency, without also increasing the max attempts, we will evict the peer much earlier than before. Increasing the delay in between attempts balances remembering the peer for a long time with also retrying it often (early on).

The current constants keeps a dead peer in the DB for 2 * 30 = 60 hours. your proposed patch will keep dead peers in the DB for 0.5 * 30 = 15 hours. My proposed constants will keep a dead peer in the DB for 0.5 * 30 + (0.1 * 30^2) / 2 = 60 hours

Reply:
tries here refers to queries on address_manager, not the tries on the selected peer (i.e. when tries=10, we made 10 queries to the address manager, giving 10 different peers). select_peer considers how many times we attempted a peer, and deprioritizes it for further selection. when tries is >= 30, it means we are going to retry peers, regardless of their last_try timestamp (but that can happen only if we have 6+ peers, and select_peer keeps picking the same peer)